File: 2/2_.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* p = l1, *q = l2;
        ListNode* res, *pre;
        ListNode* dummy_head = new ListNode(0);
        bool carry = false; //dont carry
        
        pre = dummy_head;
        while(p!=NULL || q!=NULL){
            //res *= 10;
            int digit = 0;
            digit = (p==NULL)? *q+carry : *p+*q+carry;
            digit = (q==NULL)? *p+carry : *p+*q+carry;
            //for added n addi diff digit no

            if(digi>10) {
                carry = true;
                digit -= 10;
            }
            ListNode ne_no = new ListNode(digit);
            pre->next = ne_no;

            p = p->next;
            q = q->next;
        }
            return dummy_head->next;
    }
};

File: 2/2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode dummy_head = new ListNode(0);
        ListNode pre_node = dummy_head;
        int carry = 0;

        while(l1!=null || l2!=null || carry!=0){
            int x = 0, y = 0;
            int sum = ((l2 == null) ? 0 : l2.val) + ((l1 == null) ? 0 : l1.val) + carry;
            /*
            x = (l1!=null)? l1.val : 0;
            y = (l2!=null)? l2.val : 0;
            int sum = l1 + l2 + carry;
            */
            carry = sum / 10;
            ListNode node = new ListNode(sum % 10);
            pre_node.next = node;
            pre_node = node;

            l1 = (l1 == null) ? l1 : l1.next;
            l2 = (l2 == null) ? l2 : l2.next;
        }
         return dummy_head.next;
    }
}

//Java 真難寫，真的，還醜。。
